src/lib/core.ts

```diff
@@ -72,3 +74,8 @@ That means it can't be called in a @method or similar environment, and there's n
     highBit: Bool(x >> lowBitSize === 1n),
   };
 };
+
+Scalar.fromBigInt = function (scalar: bigint) {
```

Comment:
missing helper which was useful here

Reply:
same question as elsewhere, I'm curious how a dev would know not to use these functions if they want them to create constraints? My guess is that we're warning them that going to bigint or from bigint is not constraining anything?

Reply:
For methods like these, which take a JS value and produce constant field elements, I think no warning is needed -- it's a valid and useful way to introduce constants into the circuit.

Going the other way -- from field elements to JS value -- is a very common mistake, but will throw an error as soon as you try to compile the contract. That's because extracting values, under the hood, always works like this:
```ocaml
match x with
  | Constant x -> x
  | x -> As_prover.read_var x
```
so during compile time you'll hit an error for the `As_prover.read_var x` case. The constant case is fine - no need to prove how constants are computed, since the prover can't change them